packages/oauth/oauth-provider/src/lib/http/stream.ts

```diff
@@ -73,9 +47,9 @@ export async function parseStream(
   )
 
   if (!parser) {
-    throw createHttpError(400, 'Unsupported content-type')
+    throw new TypeError(`Unsupported content-type: ${type.mime}`)
```

Comment:
Handled here:
https://github.com/bluesky-social/atproto/blob/3368726a77835d0a7197ba31239c17ea63ce2e38/packages/oauth/oauth-provider/src/lib/http/request.ts#L27

The idea is that `parseStream` is generic (does not know if its parsing a request, a response, a file on disk, etc.) so the responsibility to convert the error falls onto the caller.

Reply:
As to distinguishing null pointer error, my hypothesis is that when using TypeScript, `TypeError` are more likely invalid input errors (e.g. `new URL('ff')`) rather than actual null pointer errors.

Reply:
I could use a custom Error class for this particular purpose (because I don't want to throw something that is explicitely either a 415 or a 500). I just find `TypeError` to be handy and consistent with the way some native function work.